conftest.py

```diff
@@ -71,3 +75,41 @@ def pytest_ignore_collect(path, config):
             # If the current Python version does not meet the minimum required, skip this directory
             if sys.version_info[0:2] < min_required:
                 outcome.force_result(True)
+
+
+@pytest.fixture(scope="module")
+def docker_compose(docker_services):
+    options = {"--no-deps": False,
+               "--abort-on-container-exit": False,
+               "SERVICE": docker_services,
+               "--remove-orphans": False,
+               "--no-recreate": True,
+               "--force-recreate": False,
+               "--build": False,
+               '--no-build': False,
+               '--no-color': False,
+               "--rmi": "none",
+               "--volumes": "",
+               "--follow": False,
+               "--timestamps": False,
+               "--always-recreate-deps": False,
+               "--scale": [],
+               "--tail": "all",
+               "--detach": True,
+               "--no-log-prefix": True,
+               }
+
+    project = project_from_options(os.path.dirname(__file__), options)
+    cmd = TopLevelCommand(project)
+    cmd.up(options)
+
+    yield
+
+    cmd.logs(options)
+    cmd.down(options)
+
+
+@pytest.fixture(scope="module")
+def wait_for_testagent(docker_compose):
+    os.environ.setdefault("DD_TRACE_AGENT_URL", "http://localhost:9126")
```

Comment:
Unfortunately due to the global state in the library this won't work in all cases so we'll still need the environment variable to be defined in the test framework/CI runner.